iotdb/src/main/java/org/apache/iotdb/db/engine/filenodeV2/UnsealedTsFileProcessorV2.java

```diff
@@ -314,48 +313,55 @@ private void releaseFlushedMemTableCallback(IMemTable memTable) {
    * Take the first MemTable from the flushingMemTables and flush it. Called by a flush thread of
    * the flush manager pool
    */
-  void flushOneMemTable() throws IOException {
+  boolean flushOneMemTable() throws IOException {
     IMemTable memTableToFlush;
     memTableToFlush = flushingMemTables.getFirst();
 
     LOGGER.info("storage group {} start to flush a memtable in a flush thread", storageGroupName);
 
-    // null memtable only appears when calling asyncClose()
+    boolean flushSuccessed = false;
+    //if the memtable is not an EmptyMemTable (i.e., the memtable is actually a memtable).
     if (memTableToFlush.isManagedByMemPool()) {
       MemTableFlushTaskV2 flushTask = new MemTableFlushTaskV2(memTableToFlush, fileSchema, writer,
           storageGroupName,
           this::releaseFlushedMemTableCallback);
-      flushTask.flushMemTable();
+      flushSuccessed = flushTask.flushMemTable();
+      if (flushSuccessed) {
 //      long start = System.currentTimeMillis();
-      MemTablePool.getInstance().putBack(memTableToFlush, storageGroupName);
+        MemTablePool.getInstance().putBack(memTableToFlush, storageGroupName);
 //      long elapse = System.currentTimeMillis() - start;
 //      if (elapse > 1000) {
 //        LOGGER.info("release a memtable cost: {}", elapse);
 //      }
-      if (IoTDBDescriptor.getInstance().getConfig().isEnableWal()) {
-        getLogNode().notifyEndFlush();
+        if (IoTDBDescriptor.getInstance().getConfig().isEnableWal()) {
+          getLogNode().notifyEndFlush();
+        }
       }
       LOGGER.info("flush a memtable has finished");
     } else {
+      // the memtable is an EmptyMemTable. it is a signal for indicating asyncClose()
       LOGGER.info(
           "release an empty memtable from flushing memtable list, which is submitted in force flush");
       releaseFlushedMemTableCallback(memTableToFlush);
+      flushSuccessed = true;
     }
 
-    // for sync flush
+    // for notifying syncFlush()
     synchronized (memTableToFlush) {
       memTableToFlush.notify();
     }
 
     if (shouldClose && flushingMemTables.isEmpty()) {
-      endFile();
-
-      // for sync close
+      if (flushSuccessed) {
+        //if !flushSuccessed, then the file may be broken, we do not seal the file.
+        endFile();
+      }
+      // for notifying syncClose()
       synchronized (flushingMemTables) {
         flushingMemTables.notify();
```

Comment:
This line means that `Async Close` is done. But if `flushSuccess` is `false`, we should not tell the thread which is waiting that close is done. Maybe we need to reconsider the operation here.

Reply:
If we do not notify the waiting object, maybe they will wait forever...